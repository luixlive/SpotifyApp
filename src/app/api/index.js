import authenticationApi from './authentication';

export default authenticationApi;
