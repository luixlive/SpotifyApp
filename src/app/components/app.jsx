import PropTypes from 'prop-types';
import React from 'react';

import Header from './header';
import Footer from './footer';
import { SizeDetector } from './hoc';

export const App = props => (
  <div>
    <Header isUserAuthenticated={false} />
    {props.children}
    <Footer />
  </div>
);

App.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default SizeDetector(App);
