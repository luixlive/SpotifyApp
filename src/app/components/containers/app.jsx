import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import { Footer } from './../';
import { Header } from './../containers';
import { loadUser } from './../../actions/user';
import ScreenLoader from './../screen_loader';
import { SizeDetector } from './../hoc';

export class PureApp extends Component {
  componentDidMount() {
    this.props.loadUser();
  }

  renderBody() {
    if (this.props.userLoaded) {
      return this.props.children;
    }
    return <ScreenLoader />;
  }

  render() {
    return (
      <div>
        <Header />
        {this.renderBody()}
        <Footer />
      </div>
    );
  }
}

PureApp.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  loadUser: PropTypes.func.isRequired,
  userLoaded: PropTypes.bool.isRequired,
};

const mapStateToProps = ({ user }) => ({ userLoaded: user.userLoaded });

const mapDispatchToProps = dispatch => ({
  loadUser: () => dispatch(loadUser()),
});

export const ConnectedApp =
  connect(mapStateToProps, mapDispatchToProps)(PureApp);

export default withRouter(SizeDetector(ConnectedApp));
