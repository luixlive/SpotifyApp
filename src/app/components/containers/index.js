import App from './app';
import Error from './error';
import Footer from './footer';
import Header from './header';
import Login from './login';
import UserStats from './user_stats';

export { App, Error, Footer, Header, Login, UserStats };
