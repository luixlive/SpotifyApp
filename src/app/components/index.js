import AppConnected from './app';
import AppFooterConnected from './footer';
import AppHeaderConnected from './header';
import LoginConnected from './login';
import Router from './router';
import UserStatsConnected from './user_stats';

export {
  AppConnected,
  AppFooterConnected,
  AppHeaderConnected,
  LoginConnected,
  Router,
  UserStatsConnected,
};
