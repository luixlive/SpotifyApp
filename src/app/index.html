<!DOCTYPE html>
<html>
  <head>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-118287841-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'UA-118287841-1');
    </script>

    <!-- Favicon stuff -->
    <link rel="apple-touch-icon" sizes="180x180" href="/favicon/apple-touch-icon.png">
    <link rel="icon" type="image/png" sizes="32x32" href="/favicon/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/favicon/favicon-16x16.png">
    <link rel="manifest" href="/favicon/site.webmanifest">
    <link rel="mask-icon" href="/favicon/safari-pinned-tab.svg" color="#016936">
    <meta name="msapplication-TileColor" content="#da532c">
    <meta name="theme-color" content="#ffffff">

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Spotify App</title>
  </head>
  <body>
    <div class="app"></div>
  </body>
</html>