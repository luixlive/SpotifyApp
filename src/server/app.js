const bodyParser = require('body-parser');
const config = require('config');
const cookieSession = require('cookie-session');
const express = require('express');
const helmet = require('helmet');
const passport = require('passport');
const path = require('path');
const webpack = require('webpack');
const webpackMiddleware = require('webpack-dev-middleware');
const webpackHotMiddleware = require('webpack-hot-middleware');

const apiRouter = require('./routers');
const logger = require('./../utils/logger');
const swaggerRouter = require('./swagger');
const webpackConfig = require('./../../webpack.config');

const app = express();
const environment = config.util.getEnv('NODE_ENV');

logger.info(`Spotify App started in ${config.util.getEnv('NODE_ENV')} mode`);

require('./services/passport').configurePassport(passport);

app.use(helmet());
app.use(bodyParser.json());
app.use(cookieSession({
  maxAge: 30 * 24 * 60 * 60 * 1000,
  keys: [config.get('COOKIE_KEY')],
}));
app.use(passport.initialize());
app.use(passport.session());

app.use('/api', apiRouter);

/* istanbul ignore next  */
if (environment !== 'production') {
  app.use('/swagger', swaggerRouter);
}

const webpackCompiler = webpack(webpackConfig);
switch (environment) {
  /* istanbul ignore next */
  case 'development':
    app.use(webpackMiddleware(webpackCompiler, {}));
    app.use(webpackHotMiddleware(webpackCompiler));
    break;
  case 'test':
    app.get('/test', (req, res) => {
      res.sendStatus(200);
    });
    break;
  /* istanbul ignore next */
  default:
    app.use(express.static(path.join(__dirname, './../../dist')));
}

app.get('*', (req, res) => {
  res.sendFile(path.resolve(__dirname, './../../dist', 'index.html'));
});

module.exports = app;
