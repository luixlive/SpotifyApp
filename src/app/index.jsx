import { AppContainer } from 'react-hot-loader';
import { applyMiddleware, compose, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { Provider } from 'react-redux';
import React from 'react';
import ReactDOM from 'react-dom';

import { Router } from './components';
import reducers from './reducers';
import rootSaga from './sagas';
import './style/semantic.less';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducers,
  undefined,
  compose(
    applyMiddleware(sagaMiddleware),
    window.devToolsExtension ? window.devToolsExtension() : f => f,
  ),
);

sagaMiddleware.run(rootSaga);

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <Router />
      </Provider>
    </AppContainer>
    , document.querySelector('.app'),
  );
};

render();

if (module.hot) {
  module.hot.accept('./components/router', () => {
    render();
  });
}
