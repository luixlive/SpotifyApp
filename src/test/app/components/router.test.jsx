import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import initialState from './../../test_utils/initial_state';
import { Router } from './../../../app/components';

describe('App Components - Router', () => {
  it('renders', () => {
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const rendered = renderer.create((
      <Provider store={store}>
        <Router />
      </Provider>
    )).toJSON;
    expect(rendered).toMatchSnapshot();
  });
});
