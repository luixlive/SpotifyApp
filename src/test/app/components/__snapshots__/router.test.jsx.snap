// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Router Not Found component renders 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "marginTop": "1.5em",
      }
    }
  >
    Error 404
  </h1>
  <h2
    className="ui header"
  >
    Page not found. Go back to 
    <a
      href="/"
      onClick={[Function]}
    >
      home
    </a>
     page.
  </h2>
</div>
`;

exports[`App Components - Router renders 1`] = `
<Component>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(SizeDetector))>
        <Route
          render={[Function]}
        >
          <Connect(SizeDetector)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "blank",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "blank",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <SizeDetector
              deviceTypeChanged={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "blank",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "blank",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Connect(PureApp)
                deviceTypeChanged={[Function]}
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "blank",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "blank",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": false,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <PureApp
                  deviceTypeChanged={[Function]}
                  error=""
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "blank",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  loadUser={[Function]}
                  location={
                    Object {
                      "hash": "",
                      "pathname": "blank",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  userLoaded={false}
                >
                  <div>
                    <Connect(PureHeader)>
                      <PureHeader
                        deviceMobile={false}
                        logoutUser={[Function]}
                        userAuthenticated={false}
                      >
                        <Segment
                          color="green"
                          textAlign="center"
                        >
                          <div
                            className="ui green center aligned segment"
                          >
                            <Grid
                              columns={3}
                            >
                              <div
                                className="ui three column grid"
                              >
                                <GridColumn>
                                  <div
                                    className="column"
                                  />
                                </GridColumn>
                                <GridColumn
                                  verticalAlign="middle"
                                >
                                  <div
                                    className="middle aligned column"
                                  >
                                    <Header
                                      as="h2"
                                    >
                                      <h2
                                        className="ui header"
                                      >
                                        Spotify Stats
                                      </h2>
                                    </Header>
                                  </div>
                                </GridColumn>
                                <GridColumn>
                                  <div
                                    className="column"
                                  >
                                    <Button
                                      as="button"
                                      color="green"
                                      floated="right"
                                      inverted={true}
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="ui green inverted right floated button"
                                        onClick={[Function]}
                                        role="button"
                                      >
                                        Log in with Spotify
                                      </button>
                                    </Button>
                                  </div>
                                </GridColumn>
                              </div>
                            </Grid>
                          </div>
                        </Segment>
                      </PureHeader>
                    </Connect(PureHeader)>
                    <ScreenLoader>
                      <Segment
                        basic={true}
                        className="body"
                      >
                        <div
                          className="ui basic segment body"
                        >
                          <Dimmer
                            active={true}
                            inverted={true}
                          >
                            <div
                              className="ui active transition visible inverted dimmer"
                              onClick={[Function]}
                            >
                              <div
                                className="content"
                              >
                                <div
                                  className="center"
                                >
                                  <Loader
                                    inverted={true}
                                    size="big"
                                  >
                                    <div
                                      className="ui big inverted loader"
                                    />
                                  </Loader>
                                </div>
                              </div>
                            </div>
                          </Dimmer>
                        </div>
                      </Segment>
                    </ScreenLoader>
                    <Footer>
                      <div />
                    </Footer>
                  </div>
                </PureApp>
              </Connect(PureApp)>
            </SizeDetector>
          </Connect(SizeDetector)>
        </Route>
      </withRouter(Connect(SizeDetector))>
    </Router>
  </BrowserRouter>
</Component>
`;
